lib/dependency_spy/cli.rb

```diff
@@ -51,7 +51,10 @@ class CLI < Thor
     method_option('with-color', :type => :boolean, :default => true)
     method_option('ignore', :aliases => :i, :type => :array, :default => [])
     def check
-      manifests = API.check(options)
+      the_options = options.dup
+      the_options[:database_path] = the_options[:"database-path"]
```

Comment:
I am definitely not an expert with Ruby, but the fix seems a bit hackish since it just works for that instance.

I would prefer to go for using it like the others e.g.: `options['output-path']` or to even stop using the symbols and use strings everywhere.

What do you think?

Reply:
So since the CLI wraps the API model, I didn't want to suggest changing the API for a specific implementation (called via CLI vs importing). I figured this was cleaner as it won't break anyone/anything that already directly imports the API model.

In theory, the API model could be refactored out into its own repo and be an import if it was desired to remove all forms of implementation from the API handler.

As well, because these are relatively "fixed" arguments for the API (there are only a certain number of parameters and it's not like the keys are DB indexes), I figured using something immutable (the symbols) would be a better access method than strings themselves. (See: https://rubyinrails.com/2013/08/29/difference-between-symbols-and-strings/ and https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#keyword-arguments). There are minor performance differences between Symbols vs Strings, the former being _slightly_ faster, but that really isn't measurable in this use case.

Because the latter options are specific to the CLI implementation I felt it was fine to leave them as is. They aren't "adapting" themselves to the required arguments of another model. This PR was directly solving a bug in the implementation. If a stylistic choice is made, then I could certainly raise a PR refactoring them to conform to the decided style.

I had to `dup` the options as it's frozen when passed in via Thor.

Reply:
Fair enough I will merge it and then will think a bit better about the next steps.